raft.go

```diff
@@ -753,6 +798,19 @@ func (r *Raft) processLog(l *Log, future *logFuture) {
 // processRPC is called to handle an incoming RPC request. This must only be
 // called from the main thread.
 func (r *Raft) processRPC(rpc RPC) {
+	if v, ok := rpc.Command.(WithVersionInfo); ok {
+		if !r.isVersionCompatible(v.GetVersionInfo()) {
+			r.logger.Printf("[ERR] raft: Ignoring unsupported %#v for command: %#v", v, rpc.Command)
+			rpc.Respond(nil, ErrUnsupportedProtocol)
+			return
+		}
+
+	} else {
+		r.logger.Printf("[ERR] raft: Ignoring un-versioned command: %#v", rpc.Command)
```

Comment:
Unversioned commands come through as version 0 because of the way MsgPack decoding works - the new receivers get zero-valued version info. I updated the comment to reflect this.